### PR TITLE
Add client-benchmark-noop-api-plugin to stress clients even more in benchmarks

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this PR we add a noop plugin for Elasticsearch. It is needed to reduce the amount of request processing that is done on server-side to avoid introducing accidental bottlenecks and uncertainty on server side in order to stress clients in client-side benchmarks.

### CODE REVIEW

**Comment by rjernst** [comment]
@danielmitterdorfer Could we instead have a dummy http server (not even using ES) that responds on the expected paths? I think it would be a lot more transparent (no possibility of accidentally not setting/changing something in ES which results in real work being done).  If that doesn't work, then can we at least have a better, more descriptive name than "noop"? Something that makes it clear it is for benchmarking, and makes indexing requests noops...but I would still prefer just not using ES at all for this, since it does not seem necessary.

**Comment by danielmitterdorfer** [comment]
@rjernst I can change name, sure. Regarding the plugin's purpose. We could use a dummy HTTP server but the thing is that this is also used for benchmarking the transport client and there is no substitute for that. Also, using ES for the server-side component ensures that the interaction model stays as close as possible to a real use case.

**Comment by danielmitterdorfer** [comment]
I've renamed the plugin now to "client-benchmark-noop-api-plugin"

**Comment by danielmitterdorfer** [comment]
@rjernst Are you otherwise fine with the change (especially after my explanation why the plugin is needed at all)?

File: client/benchmark/src/main/java/org/elasticsearch/client/benchmark/AbstractBenchmark.java [hunk 1]
```diff
@@ -40,54 +44,113 @@
...
-        if (args.length < 6) {
+        if (args.length < 1) {
+            System.err.println("usage: [search|bulk]");
+            System.exit(1);
+        }
+        switch (args[0]) {
+            case "search":
+                runSearchBenchmark(args);
+                break;
+            case "bulk":
```
Comment (rjernst) [inline client/benchmark/src/main/java/org/elasticsearch/client/benchmark/AbstractBenchmark.java]: just curious, why isn't this called "indexing"?

**Comment by rjernst** [comment]
LGTM

File: client/benchmark/src/main/java/org/elasticsearch/client/benchmark/AbstractBenchmark.java (hunk 1 above)
Comment (danielmitterdorfer) [inline client/benchmark/src/main/java/org/elasticsearch/client/benchmark/AbstractBenchmark.java]: No special reason tbh. I just used the specific indexing approach as name.

**Comment by danielmitterdorfer** [comment]
Thanks for your review Ryan.

**Pull request merged**